pkg/appfile/appfile.go

```diff
@@ -269,16 +293,27 @@ func generateComponentFromCUEModule(wl *Workload, appName, revision, ns string)
 }
 
 // evalWorkloadWithContext evaluate the workload's template to generate component and ACComponent
-func evalWorkloadWithContext(pCtx process.Context, wl *Workload, appName, compName string) (*v1alpha2.Component, *v1alpha2.ApplicationConfigurationComponent, error) {
+func evalWorkloadWithContext(pCtx process.Context, wl *Workload, ns, appName, compName string) (*v1alpha2.Component, *v1alpha2.ApplicationConfigurationComponent, error) {
+	var (
+		commonLabels      map[string]string
+		componentWorkload *unstructured.Unstructured
+		err               error
+	)
 	base, assists := pCtx.Output()
-	componentWorkload, err := base.Unstructured()
-	if err != nil {
-		return nil, nil, errors.Wrapf(err, "evaluate base template component=%s app=%s", compName, appName)
+	switch wl.CapabilityCategory {
+	case types.TerraformCategory:
+		componentWorkload, err = generateTerraformConfigurationWorkload(wl, ns)
+		if err != nil {
+			return nil, nil, errors.Wrapf(err, "failed to generate Terraform Configuration workload for workload %s", wl.Name)
+		}
+	default:
+		componentWorkload, err = base.Unstructured()
+		if err != nil {
+			return nil, nil, errors.Wrapf(err, "evaluate base template component=%s app=%s", compName, appName)
+		}
+		commonLabels = definition.GetCommonLabels(pCtx.BaseContextLabels())
+		util.AddLabels(componentWorkload, util.MergeMapOverrideWithDst(commonLabels, map[string]string{oam.WorkloadTypeLabel: wl.Type}))
 	}
-
-	var commonLabels = definition.GetCommonLabels(pCtx.BaseContextLabels())
-	util.AddLabels(componentWorkload, util.MergeMapOverrideWithDst(commonLabels, map[string]string{oam.WorkloadTypeLabel: wl.Type}))
```

Comment:
we should also add lables for terrarform?

Reply:
Done.